docs/md_merge.py

```diff
@@ -0,0 +1,357 @@
+    for f in sorted(files):
+        if f not in result:
+            result.append(f)
+
+    return result
+
+
+LANGUAGE_MD_FILE_NAME_PATTERN = "(.+)[.](.{2,3})[.]md"
+
+
+def get_md_language_files(dir_path: Path, language: str = "en"):
+    """tries to fetch all markdown files with a language specific suffix
+    assumes that files in English (suffix '.en.md') are all present
+    and uses these as fallback"""
+    lang_to_file_prefix = defaultdict(list)
+    for md_file in os.listdir(dir_path):
+        match = re.match(LANGUAGE_MD_FILE_NAME_PATTERN, md_file)
+        if match:
+            lang = match.group(2)
+            lang_to_file_prefix[lang].append(match.group(1))
+    result = []
+    for prefix in lang_to_file_prefix["en"]:
+        expected_file = prefix + f".{language}.md"
+        if prefix in lang_to_file_prefix[language]:
+            result.append(expected_file)
+        else:
+            print(f"missing file '{expected_file}', falling back to English version")
+            result.append(prefix + ".en.md")
+    return result
+
+
+LINK_COUNT = 1
+IMG_COUNT = 1
+PATH_TO_DEPTH_AND_CONTENT = OrderedDict()
+PATH_TO_LINK_ID = defaultdict(list)
+
+
+def parse_dir_recursive(
+        start_dir: Path,
+        current_dir: Path,
+        output_dir: Path,
+        language: str,
+        depth=1,
+        max_depth=-1,
+        paths_relative_to_input_dir=False
+):
+    global LINK_COUNT
+    global IMG_COUNT
+
+    files_only = max_depth == 0
+
+    dir_content = get_dir_content(current_dir, language, files_only=files_only)
+    for dc in dir_content:
+        path = current_dir / dc
+        if path.is_file():
+            file_lines = []
+            for line in path.read_text().splitlines():
+                parsed_line = parse_line(line)
+                if isinstance(parsed_line, ImageLine) and not parsed_line.is_external:
+                    new_image_name = f"md-merge-img-{IMG_COUNT}-" + Path(parsed_line.link).name
+                    new_image_path = output_dir / "images" / new_image_name
+                    shutil.copyfile(current_dir / parsed_line.link, new_image_path)
+                    parsed_line.link = str(new_image_path.relative_to(output_dir))
+                if isinstance(parsed_line, TextLinkLine) and not (parsed_line.is_external
+                                                                  or parsed_line.is_anchor_link):
+                    if paths_relative_to_input_dir:
+                        linked_path = start_dir / parsed_line.link
+                    else:
+                        linked_path = path.parent / parsed_line.link
+                    link_name = f"md-merge-link-{LINK_COUNT}"
+                    parsed_line.link = "#" + link_name
+                    PATH_TO_LINK_ID[linked_path].append(link_name)
+                    LINK_COUNT += 1
+                file_lines.append(parsed_line)
+            PATH_TO_DEPTH_AND_CONTENT[path] = (depth, file_lines)
+        elif path.is_dir():
```

Comment:
Nit: Confusing mix the other way around, here we use the super power path lib functionality to test for a "path" being a directory ... in my daily work this sometimes happens, when the developers copy functions together until the assembly does not fail tests anymore ... just saying, not meaning to sound harsh.

Reply:
Updated to get rid of `os` altogether.